recon/src/recon.rs

```diff
@@ -87,8 +86,12 @@ where
     }
 
     /// Process an incoming message and respond with a message reply.
+    /// Return Result<(response_message, synced_count, syncing_count), Error>
     #[instrument(skip_all, ret)]
-    pub async fn process_messages(&mut self, received: &[Message<K, H>]) -> Result<Response<K, H>> {
+    pub async fn process_messages(
+        &mut self,
+        received: &[Message<K, H>],
+    ) -> Result<(Response<K, H>, u64, u64)> {
```

Comment:
Can we add synced and syncing  on the Response struct itself? Response does not go across the wire so we can add progress metadata like this to it.